src/server/req_delete.c

```diff
@@ -343,88 +382,54 @@ req_deletejob(struct batch_request *preq)
 		qdel_mail = 0;
 	else
 		qdel_mail = 1;
-	parent = chk_job_request(jid, preq, &jt);
-	if (parent == NULL)
+
+	if (FALSE == chk_job_request(jid, preq, &jt, &js, &pjob, &parent, &offset))
 		return; /* note, req_reject already called */
 
 	if (delhist) {
-		rc = check_deletehistoryjob(preq);
+		rc = check_deletehistoryjob(preq, pjob, parent, jt);
 		if (rc == TRUE)
 			return;
 	}
 
 
-	if (jt == IS_ARRAY_NO) {
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
 
-		/* just a regular job, pass it on down the line and be done
+		/* just a regular job or a subjob, pass it on down the line and be done
 		 * If the request if to purge the history of the job then set ji_deletehistory to 1
```

Comment:
Change the typo if the request if to purge to if the request is to purge

Reply:
Fixed in rc 16 https://github.com/PBSPro/pbspro/pull/590/commits/bc13b0be5a010f69357b3cde4fccafea0e0fb92d